hles/transformation/src/main/scala/org/broadinstitute/monster/dap/RawRecord.scala

```diff
@@ -50,7 +53,30 @@ case class RawRecord(id: Long, fields: Map[String, Array[String]]) {
   def getOptionalBoolean(field: String): Option[Boolean] = getOptional(field).map(_ == "1")
 
   /** Get the singleton value for an attribute in this record, parsed as a long. */
-  def getOptionalNumber(field: String): Option[Long] = getOptional(field).map(_.toLong)
+  def getOptionalNumber(field: String, truncateDecimals: Boolean = true): Option[Long] =
+    getOptional(field).map(value => {
+      try {
+        value.toLong
```

Comment:
🔧 I think the python-ish EAFP pattern should be avoided in Java. Exceptions are expensive operations and should be avoided for control flow purposes. What if we check `truncateDecimals` first and if not set only then do the `toLong`? That way we can avoid the `try/catch` altogether and let the exception bubble up to the caller unchanged.